ui/apps/platform/src/Containers/Violations/Details/DeploymentDetails.tsx

```diff
@@ -1,11 +1,13 @@
 import React, { useEffect, useState } from 'react';
 import { Alert, Flex, FlexItem, Card, CardBody, Title, Divider } from '@patternfly/react-core';
+import { TableComposable, Caption, Thead, Tr, Th, Tbody, Td } from '@patternfly/react-table';
 
-import { fetchDeployment } from 'services/DeploymentsService';
 import { fetchNetworkPoliciesInNamespace } from 'services/NetworkService';
 import { portExposureLabels } from 'messages/common';
 import ObjectDescriptionList from 'Components/ObjectDescriptionList';
-import { TableComposable, Caption, Thead, Tr, Th, Tbody, Td } from '@patternfly/react-table';
+import useFetchDeployment from 'hooks/useFetchDeployment';
+import { getAxiosErrorMessage } from 'utils/responseErrorUtils';
+import { Alert as AlertViolation } from '../types/violationTypes';
```

Comment:
Possible follow up, although maybe beyond scope of your involvement here:
* supersede `Alert` in container: https://github.com/stackrox/stackrox/blob/master/ui/apps/platform/src/Containers/Violations/types/violationTypes.ts#L114-L137
* with supposed source of truth: https://github.com/stackrox/stackrox/blob/master/ui/apps/platform/src/types/alert.proto.ts#L133-L165

Although not taking a side on trust issues, container type has optional chaining where frontend proto explicitly distinguishes subtypes, therefore allows for type-safe use for corresponding conditional code.

Reply:
It looks like `Alert` and `ListAlert` are both sources of truth, the backend returns the regular `Alert` when hitting the single entity API: https://github.com/stackrox/stackrox/blob/master/proto/api/v1/alert_service.proto#L115

There are some inconsistencies though that I haven't dug into (e.g. `Alert` in the UI declares an optional `commonEntityInfo` field, which the backend doesn't declare at all and isn't in the response). If the type isn't too tangled in the logic, a good follow up would be to straighten out these properties and move `Alert` to the global `types` dir in our repo. I'll make a note and add it to the nice to haves list.